### PR TITLE
[Transaction Manager] Nonce Too High Error and more example clarity.

### PR DESCRIPTION
**Pull request by arnav-deep**
### Nonce Too High Error
- Ensure any nonce set for a transaction is utilised, even if sending that transaction fails using a dummy transaction.
- Add Gas Price and Qty Estimation in dummy transaction.

### Example Clarity
- ContractInstance of `alloy-rs` can be used to create function call struct and converted into bytes to get the transaction data for that fn call.

### Contract Execution Error while Gas Estimation
- ContractExecution error during gas estimation does not stop the fn call and keeps on trying. Stop retries and send a dummy transaction to utilise nonce.

### CODE REVIEW

**Review by vg-27 — COMMENTED** [review]
Changes looks good. Need testing results before merging.

File: serverless/transaction-manager/src/utils.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@ pub(crate) fn parse_send_error(error: String) -> TxnManagerSendError {
     if error_lowercase.contains("out of gas")
         || (error_lowercase.contains("transaction requires at least")
             && error_lowercase.contains("gas but got"))
+        || error_lowercase.contains("gas too low")
```
Comment (vg-27) [inline serverless/transaction-manager/src/utils.rs:21]: is this  is the error that you got from sepolia rpc?

File: serverless/transaction-manager/src/utils.rs [hunk 2]
```diff
@@ -18,6 +18,7 @@ pub(crate) fn parse_send_error(error: String) -> TxnManagerSendError {
     if error_lowercase.contains("out of gas")
         || (error_lowercase.contains("transaction requires at least")
```
Comment (vg-27) [inline serverless/transaction-manager/src/utils.rs:19]: btw, I noticed this, this error occurs when, when gas account has less ETHs than required, am I right? If yes, then we can remove it from here and create another error type.

**Review by vg-27 — COMMENTED** [review]
LG

File: serverless/transaction-manager/src/utils.rs (hunk 1 above)
Comment (arnav-deep) [reply serverless/transaction-manager/src/utils.rs:21]: Yes

File: serverless/transaction-manager/src/utils.rs (hunk 2 above)
Comment (arnav-deep) [reply serverless/transaction-manager/src/utils.rs]: Updated to a new type

File: serverless/transaction-manager/src/errors.rs [hunk 3]
```diff
@@ -12,6 +12,8 @@ pub enum TxnManagerSendError {
     NonceTooHigh(String),
     #[error("Out of gas. Error: {0}")]
     OutOfGas(String),
+    #[error("Not enough gas. Error: {0}")]
+    NotEnoughGas(String),
```
Comment (vg-27) [inline serverless/transaction-manager/src/errors.rs:16]: can we name it `InsufficientBalance` ?

File: serverless/transaction-manager/src/errors.rs [hunk 4]
```diff
@@ -13,7 +13,7 @@ pub enum TxnManagerSendError {
     #[error("Out of gas. Error: {0}")]
     OutOfGas(String),
     #[error("Not enough gas. Error: {0}")]
```
Comment (vg-27) [inline serverless/transaction-manager/src/errors.rs:15]: update the string too

**Review by vg-27 — APPROVED** [review]
LG

File: serverless/transaction-manager/src/errors.rs (hunk 4 above)
Comment (arnav-deep) [reply serverless/transaction-manager/src/errors.rs]: Done

**Review by vg-27 — APPROVED** [review]
LG

**Pull request merged**